ARCharts/ARBarChart.swift

```diff
@@ -125,21 +129,31 @@ public class ARBarChart: SCNNode {
                 let value = dataSource.barChart(self, valueAtIndex: index, forSeries: series)
                 
                 let barHeight = Float(value) * maxBarHeight
+                let startingBarHeight = animationType == .grow || animationType == .progressiveGrow ? 0.0 : barHeight
                 let barBox = SCNBox(width: CGFloat(barsLength),
-                                    height: CGFloat(barHeight),
+                                    height: CGFloat(startingBarHeight),
                                     length: CGFloat(seriesSize),
                                     chamferRadius: 0)
                 let barNode = SCNNode(geometry: barBox)
+                let opacity = delegate.barChart(self, opacityForBarAtIndex: index, forSeries: series)
+                let startingOpacity = animationType == .fadeIn || animationType == .progressiveFadeIn ? 0.0 : opacity
+                barNode.opacity = CGFloat(startingOpacity)
                 
-                let yPosition = Float(value) * Float(maxBarHeight) / 2.0
+                let yPosition = Float(value) * Float(maxBarHeight) / 2
+                let startingYPosition = animationType == .grow || animationType == .progressiveGrow ? 0.0 : yPosition
                 let xPosition = self.xPosition(forIndex: index, previousXPosition, barsLength)
-                barNode.position = SCNVector3(x: xPosition, y: yPosition, z: zPosition)
+                barNode.position = SCNVector3(x: xPosition, y: startingYPosition, z: zPosition)
                 
                 let barColor = delegate.barChart(self, colorForBarAtIndex: index, forSeries: series)
                 barNode.geometry?.firstMaterial?.diffuse.contents = barColor
                 
                 self.addChildNode(barNode)
                 previousXPosition = xPosition
+                
+                if let animationType = animationType {
+                    let animationManager = AnimationManager(animationType: animationType, animationDuration: animationDuration)
```

Comment:
Can we construct the `AnimationManager` only once on the bar chart, when the `animationType` gets set?